src/azure-cli/azure/cli/command_modules/rdbms/_params.py

```diff
@@ -80,6 +80,8 @@ def _complex_params(command_group):
             c.argument('administrator_login', options_list=['--admin-user', '-u'], help='Administrator username for the server. Once set, it cannot be changed.')
             c.argument('administrator_login_password', options_list=['--admin-password', '-p'], help='The password of the administrator. Minimum 8 characters and maximum 128 characters. Password must contain characters from three of the following categories: English uppercase letters, English lowercase letters, numbers, and non-alphanumeric characters.')
             c.argument('ssl_enforcement', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--ssl-enforcement'], help='Enable or disable ssl enforcement for connections to server. Default is Enabled.')
+            c.argument('minimal_tls_version', arg_type=get_enum_type(['TLS1_0', 'TLS1_1', 'TLS1_2', 'TLSEnforcementDisabled']), options_list=['--minimal-tls-version'], help='Set the minimal TLS version for connections to server when SSL is enabled. Default is TLS1_2.', validator=tls_validator)
+            c.argument('public_network_access', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--public-network-access'], help='Set whether or not public network access is allowed for this server. Default is Enabled')
```

Comment:
Can you change to use get_three_state_flags() like https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/sql/_params.py#L1048?

Reply:
All three services covered in the rdbms (mysql, postgres and mariadb uses different model though same) hence changing this to use get_three_state_flags will add additional conversions for each of these types separately. We also have ssl enforcement which has the same problem hence we have kept it as enabled and disabled for now. Will think about this to see how we can make a generic change.